runtime/moonbase/src/bridge_config.rs

```diff
@@ -0,0 +1,310 @@
+use frame_support::pallet_prelude::PalletInfoAccess;
+use frame_support::traits::{Contains, EnqueueMessage, Everything};
+use frame_support::{ensure, parameter_types, traits::ConstU32, BoundedVec};
+use frame_system::{EnsureNever, EnsureRoot};
+use moonbeam_core_primitives::{AccountId, Balance};
+use pallet_xcm_bridge::congestion::{
+	BlobDispatcherWithChannelStatus, HereOrLocalConsensusXcmChannelManager,
+	UpdateBridgeStatusXcmChannelManager,
+};
+use pallet_xcm_bridge::XcmAsPlainPayload;
+use parity_scale_codec::{Decode, Encode};
+use polkadot_parachain::primitives::Sibling;
+use sp_core::hex2array;
+use sp_runtime::traits::Convert;
+use sp_runtime::Vec;
+use sp_std::vec;
+use sp_weights::Weight;
+use xcm::latest::{AssetId, Junction, Location, MaybeErrorCode, OriginKind, Xcm};
+use xcm::opaque::VersionedXcm;
+use xcm::prelude::{
+	ExpectTransactStatus, GlobalConsensus, InteriorLocation, NetworkId, PalletInstance, Parachain,
+	Transact, Unlimited, UnpaidExecution,
+};
+use xcm_builder::{
+	BridgeMessage, DispatchBlob, DispatchBlobError, ParentIsPreset, SiblingParachainConvertsVia,
+};
+
+pub struct LocalBlobDispatcher<MQ, OurPlace, OurPlaceBridgeInstance>(
+	PhantomData<(MQ, OurPlace, OurPlaceBridgeInstance)>,
+);
+impl<
+		MQ: EnqueueMessage<AggregateMessageOrigin>,
+		OurPlace: Get<InteriorLocation>,
+		OurPlaceBridgeInstance: Get<Option<InteriorLocation>>,
+	> DispatchBlob for LocalBlobDispatcher<MQ, OurPlace, OurPlaceBridgeInstance>
+{
+	fn dispatch_blob(blob: Vec<u8>) -> Result<(), DispatchBlobError> {
+		let our_universal = OurPlace::get();
+		let our_global = our_universal
+			.global_consensus()
+			.map_err(|()| DispatchBlobError::Unbridgable)?;
+		let BridgeMessage {
+			universal_dest,
+			message,
+		} = Decode::decode(&mut &blob[..]).map_err(|_| DispatchBlobError::InvalidEncoding)?;
+		let universal_dest: InteriorLocation = universal_dest
+			.try_into()
+			.map_err(|_| DispatchBlobError::UnsupportedLocationVersion)?;
+		// `universal_dest` is the desired destination within the universe: first we need to check
+		// we're in the right global consensus.
+		let intended_global = universal_dest
+			.global_consensus()
+			.map_err(|()| DispatchBlobError::NonUniversalDestination)?;
+		ensure!(
+			intended_global == our_global,
+			DispatchBlobError::WrongGlobal
+		);
+		//let dest = universal_dest.relative_to(&our_universal);
+		let xcm: Xcm<()> = message
+			.try_into()
+			.map_err(|_| DispatchBlobError::UnsupportedXcmVersion)?;
+
+		let msg: BoundedVec<u8, MQ::MaxMessageLen> = VersionedXcm::from(xcm)
+			.encode()
+			.try_into()
+			.map_err(|_| DispatchBlobError::InvalidEncoding)?;
+		MQ::enqueue_message(
+			msg.as_bounded_slice(),
+			AggregateMessageOrigin::Here, // The message came from the para-chain itself.
+		);
+
+		Ok(())
+	}
+}
+
+/// Converts encoded call to the unpaid XCM `Transact`.
+pub struct UpdateBridgeStatusXcmProvider;
+impl Convert<Vec<u8>, Xcm<()>> for UpdateBridgeStatusXcmProvider {
```

Comment:
Does this mean that we allow any substrate call via the bridge?
I think it's better to allow only EthereumXcm call and "maintenance" call (and perform some checks against the origin for the later)